src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2112,6 +2098,12 @@ async def lock(self, id, options):
         elif id == (await self.middleware.call('systemdataset.config'))['pool']:
             raise CallError(f'Please move system dataset to another pool before locking {id}')
 
+        async def detach(delegate):
+            await delegate.detach((await delegate.query(self.__attachments_path(ds), True, {'locked': False})))
```

Comment:
`detach` either needs to be no-op if it has no attachments (now it restarts service anyway) or we should not call it if there are no attachments. Same for `stop` and `start`.